mobile/src/libs/hooks/use-search/use-search.hook.ts

```diff
@@ -0,0 +1,24 @@
+import { type Dispatch, type SetStateAction } from 'react';
+
+import { useState } from '#libs/hooks/hooks';
+
+import { DEFAULT_SEARCH_PAYLOAD } from './libs/constants';
+
+type UseSearchResult<T> = {
+  filteredData: T[];
+  setSearchQuery: Dispatch<SetStateAction<string>>;
+};
+
+const useSearch = <T>(data: T[], propertyName: keyof T): UseSearchResult<T> => {
+  const [searchQuery, setSearchQuery] = useState(DEFAULT_SEARCH_PAYLOAD.search);
```

Comment:
Just store filteredData as a state. searchQuery already stored in form

Reply:
I've added filteredData to state. Could you please check?